### PR TITLE
fix(sysadvisor): fix failed to find regions and pools

### PR DESCRIPTION
**Pull request by cheney-lin**
#### What type of PR is this?

Bug fixes

### CODE REVIEW

**Review by sun-yuliang — DISMISSED** [review]

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/regulator/regulator.go [hunk 1]
```diff
@@ -96,6 +96,7 @@ func (c *CPURegulator) slowdown(cpuRequirement float64) float64 {
 	// Restrict ramp down period
 	if cpuRequirement < latestCPURequirement && now.Before(c.latestRampDownTime.Add(c.minRampDownPeriod)) {
 		cpuRequirement = latestCPURequirement
```
Comment (waynepeking348) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/region/regulator/regulator.go:98]: return latestCPURequirement; it seems we don't need to update to cpuRequirement

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go [hunk 2]
```diff
@@ -491,15 +491,39 @@ func (cra *cpuResourceAdvisor) gc() {
 	}
 }
 
-func (cra *cpuResourceAdvisor) getContainerRegions(ci *types.ContainerInfo) ([]region.QoSRegion, error) {
+func (cra *cpuResourceAdvisor) getRegionsByRegionNames(names sets.String) []region.QoSRegion {
 	var regions []region.QoSRegion = nil
-	for regionName := range ci.RegionNames {
-		r, ok := cra.regionMap[regionName]
+	for regionName := range names {
+		region, ok := cra.regionMap[regionName]
```
Comment (waynepeking348) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:497]: region is the same with the import name, change this to r

File: pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go [hunk 3]
```diff
@@ -491,15 +491,39 @@ func (cra *cpuResourceAdvisor) gc() {
...
+				regions = append(regions, r)
+			}
+		}
+	}
+	return regions
+}
+
+func (cra *cpuResourceAdvisor) getContainerRegions(ci *types.ContainerInfo) ([]region.QoSRegion, error) {
+	regions := cra.getRegionsByRegionNames(ci.RegionNames)
+	if len(regions) > 0 {
```
Comment (waynepeking348) [inline pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:521]: when will `getRegionsByRegionNames` returns nil, but `getRegionsByPodUID` has data?

Comment (cheney-lin) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:521]: When a new container(sidecar as example) allocated, its regionNames are empty, so `getRegionsByRegionNames` will returns nil. But its main continer has already assigned to specified region, so `getRegionsByPodUID` returns the regions of main container

Comment (waynepeking348) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:521]: okay, got it, maybe we'd better add those above as comments to clarify

Comment (cheney-lin) [reply pkg/agent/sysadvisor/plugin/qosaware/resource/cpu/advisor.go:523]: done

**Review by waynepeking348 — APPROVED** [review]

**Review by sun-yuliang — APPROVED** [review]

**Pull request merged**